frontend/src/pages/NewRunSwitcher.tsx

```diff
@@ -43,7 +45,36 @@ function NewRunSwitcher(props: PageProps) {
     },
     { enabled: !!existingRunId, staleTime: Infinity },
   );
-  const templateStrFromRunId = apiRun ? apiRun.run?.pipeline_spec?.pipeline_manifest : '';
+
+  const {
+    isSuccess: getRecurringRunSuccess,
+    isFetching: recurringRunIsFetching,
+    data: apiRecurringRun,
+  } = useQuery<ApiJob, Error>(
+    ['ApiRecurringRun', originalRecurringRunId],
+    () => {
+      if (!originalRecurringRunId) {
+        throw new Error('Recurring Run ID is missing');
+      }
+      return Apis.jobServiceApi.getJob(originalRecurringRunId);
+    },
+    { enabled: !!originalRecurringRunId, staleTime: Infinity },
+  );
+
+  let pipelineManifestFromRecurringRun: string | undefined;
+  let pipelineManifestFromRun: string | undefined;
+
+  if (getRecurringRunSuccess && apiRecurringRun) {
+    pipelineManifestFromRecurringRun = apiRecurringRun.pipeline_spec?.pipeline_manifest;
+  }
+
+  if (getRunSuccess && apiRun) {
+    pipelineManifestFromRun = apiRun.run?.pipeline_spec?.pipeline_manifest;
+  }
+
+  let pipelineManifest = pipelineManifestFromRun
+    ? pipelineManifestFromRun
+    : pipelineManifestFromRecurringRun;
```

Comment:
The logics for determining `templateString` has been scattered across multiple places in this file (here and below). Suggest to put all the `API fetching` logics in one session, and combine all `templateString` related logics at the end.